### PR TITLE
add queryable capability to handles with refinements

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/runtime/capabilities.ts [hunk 1]
```diff
@@ -479,6 +479,16 @@ export class Capabilities {
     return new Capabilities(ranges);
   }
 
+  setCapability(capability: Capability, type: typeof Capability): boolean {
```
Comment (mmandlis) [inline src/runtime/capabilities.ts:482]: this is really dumb, but i cannot find a way to get `typeof` out of the Capability object :(

Comment (csilvestrini) [reply src/runtime/capabilities.ts:482]: The only thing I can think of is to use a tag for all Capability subtypes. e.g. something like this:

```
type CapabilityTag = 'persistence' | 'ttl' | ...;

class Capability {
  abstract readonly tag: CapabilityTag;
}

class Persistence extends Capability {
  tag: 'persistence';
  ...
}

  setCapability(capability: Capability) {
     let existing = this.getWithTag(capablity.tag);
     ...
  }
```

Comment (mmandlis) [reply src/runtime/capabilities.ts]: i really like this, done. thanks!

Comment (mmandlis) [reply src/runtime/capabilities.ts]: I'll merge now, but let me know if you have comments and i'll address them in a follow up PR.

Comment (csilvestrini) [reply src/runtime/capabilities.ts]: LGTM!

**Pull request merged**